### PR TITLE
FLA-286 Retrieve Court Descriptions from CSO Part 1

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

implement soap service get description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Unit test

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/service/EfilingCourtService.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package ca.bc.gov.open.jag.efilingcommons.service;
+
+import ca.bc.gov.open.jag.efilingcommons.model.CourtDetails;
+
+public interface EfilingCourtService {
```
Comment (alexjoybc) [inline src/backend/libs/efiling-commons/src/main/java/ca/bc/gov/open/jag/efilingcommons/service/EfilingCourtService.java:5]: nice

File: src/backend/libs/efiling-cso-starter/src/main/java/ca/bc/gov/open/jag/efilingcsostarter/CsoCourtServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public CourtDetails getCourtDescription(String agencyIdentifierCd) {
+        if (StringUtils.isBlank(agencyIdentifierCd)) throw new IllegalArgumentException("Agency identifier is required");
+
+        CsoAgencyArr csoAgencyArr = csows.getCourtLocations();
+        Optional<CsoAgencyRec> csoAgencyRec =  csoAgencyArr.getArray().stream()
+                .filter(court -> court.getAgenAgencyIdentifierCd().equals(agencyIdentifierCd))
+                .findFirst();
```
Comment (alexjoybc) [inline src/backend/libs/efiling-cso-starter/src/main/java/ca/bc/gov/open/jag/efilingcsostarter/CsoCourtServiceImpl.java:29]: do we want to do the `orElseThrow` ?

File: src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoCourtService/GetCourtDescription.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        CsoAgencyRec csoAgencyRec2 = new CsoAgencyRec();
+        csoAgencyRec2.setAgenAgencyIdentifierCd(AGEN_AGENCY_IDENTIFIER_CD1);
+        csoAgencyRec2.setAgenAgencyNm(AGEN_AGENCY_NM1);
+        csoAgencyRec2.setAgenId(BigDecimal.TEN);
+        Mockito.when(csoAgencyArrMock.getArray()).thenReturn(Arrays.asList(csoAgencyRec1, csoAgencyRec2));
+        Mockito.when(csowsMock.getCourtLocations()).thenReturn(csoAgencyArrMock);
+
+        sut = new CsoCourtServiceImpl(csowsMock);
+    }
+    @DisplayName("Exception: with null paramtere throws Illegal Argument exception")
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoCourtService/GetCourtDescription.java:53]:
```suggestion
    @DisplayName("Exception: with null parameter throws Illegal Argument exception")
```

File: src/backend/libs/efiling-demo-starter/src/main/java/ca/bc/gov/open/jag/efiling/demo/EfilingCourtServiceDemoImpl.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import ca.bc.gov.open.jag.efilingcommons.model.CourtDetails;
+import ca.bc.gov.open.jag.efilingcommons.service.EfilingCourtService;
+
+import java.math.BigDecimal;
+
+public class EfilingCourtServiceDemoImpl implements EfilingCourtService {
+    @Override
+    public CourtDetails getCourtDescription(String agencyIdentifierCd) {
+        return new CourtDetails(BigDecimal.TEN, "Imma Court");
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-demo-starter/src/main/java/ca/bc/gov/open/jag/efiling/demo/EfilingCourtServiceDemoImpl.java:11]: 😸

File: src/backend/libs/efiling-cso-starter/src/main/java/ca/bc/gov/open/jag/efilingcsostarter/CsoCourtServiceImpl.java (hunk 2 above)
Comment (TayGov) [reply src/backend/libs/efiling-cso-starter/src/main/java/ca/bc/gov/open/jag/efilingcsostarter/CsoCourtServiceImpl.java]: Good catch

**Pull request merged**